Written tests, scripts and refactored the TREE NFT contract

IronJam11 opened the pull request:
### **PR Description: Smart Contract Implementation for Tree NFT Project**

This PR introduces the following key deliverables for the Tree NFT project:

### 🚀 **Deliverables:**
1. **Smart Contract:**  
   - Rewritten the entire smart contract to include verifier logic for validating tree planting claims before minting NFTs.  

2. **Testing:**  
   - Comprehensive tests written for the following functionalities:  
     - Verification of claims  
     - Deployment of the smart contract  
     - NFT minting process  

3. **Deployment & Minting Scripts:**  
   - Created scripts for:  
     - Smart contract deployment  
     - Minting NFTs after successful verification  

---

### 🛠️ **How to Run:**

1. **Environment Setup:**  
   - Create a `.env` file with the following variables:  
     ```bash
     SEPOLIA_RPC_URL=<your_sepolia_rpc_url>
     PRIVATE_KEY=<your_private_key>
     ETHERSCAN_API_KEY=<your_etherscan_api_key>
     ```
   - Source the environment variables:  
     ```bash
     source .env
     ```

2. **Install Dependencies:**  
   - Install the required packages without committing them:  
     ```bash
     forge install ChainAccelOrg/foundry-devops --no-commit
     forge install OpenZeppelin/openzeppelin-contracts --no-commit
     ```

3. **Deployment & Minting:**  
   - To directly mint an NFT using the deployment script:  
     ```bash
     forge script script/MintTreeNft.s.sol --rpc-url $SEPOLIA_RPC_URL --broadcast --verify -vvvv
     ```

4. **Optional Deployment:**  
   - Alternatively, you can deploy the smart contract using:  
     ```bash
     make deploy ARGS="--network sep"
     ```

---
<img width="350" alt="image" src="https://github.com/user-attachments/assets/19369ec1-45c1-454e-97e7-8b65edaabf42" />

ceilican commented on src/TreeeNft.sol:
```diff
@@ -0,0 +1,125 @@
+    struct Tree {
+        uint256 latitude;
+        uint256 longitude;
+        uint256 data;
```
`data` => `planting // block timestamp when the tree was planted`

ceilican commented on src/TreeeNft.sol:
```diff
@@ -0,0 +1,125 @@
+        uint256 data;
+    }
+
+    string private s_treeSvgImageUri;
```
What is this variable for?

IronJam11 commented on src/TreeeNft.sol:
```diff
@@ -0,0 +1,125 @@
+        uint256 latitude;
+        uint256 longitude;
+        uint256 data;
+    }
```
Okay I’ll make the necessary changes

IronJam11 commented:
Added the following changes:- 
- Added dead, planting, imageURI to the struct 
- Modified the tests 
- Removed unnecessary variables 
- Added markDead function
